src/jit/instrxarch.h

```diff
@@ -0,0 +1,107 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifndef _INSTR_XARCH_H_
+#define _INSTR_XARCH_H_
+
+#if !defined(_TARGET_XARCH_)
+#error Unexpected target type
+#endif
+
+enum InsEncodingFmt : unsigned int
+{
+    // "R/M, [reg]" addressing mode
+    InsEncodingFmt_mr,
+
+    // "R/M, icon" addressing mode
+    InsEncodingFmt_mi,
+
+    // "reg, R/M" addression mode
+    InsEncodingFmt_rm,
+
+    // "eax, i32" addressing mode
+    InsEncodingFmt_a4,
+
+    // "reg, reg" addressing mode
+    InsEncodingFmt_rr,
+
+    InsEncodingFmt_none,
+    InsEncodingFmt_count = InsEncodingFmt_none
+};
+
+enum InsFlag : unsigned int
+{
+    InsFlag_None = 0,
+
+    // Floating point instructions
+    InsFlag_FloatingPoint = 1,
+
+    // Reads flags
+    InsFlag_ReadsFlags = 2,
+
+    // Writes flags
+    InsFlag_WritesFlags = 4,
+};
+
+struct InstructionInfo
+{
+#if DEBUG
+    instruction id;
+    const char* name;
+#endif
+
+    size_t         encoding[5];
+    insUpdateModes updateMode;
```

Comment:
This is wasting 3 bytes per entry, since `insUpdateModes` is an enum that defaults to `int` type. Why isn't this part of `flags`?

Reply:
It could be changed to be two flags (UpdateModeRead and UpdateModeWrite) if desired.